src/main/java/reactor/core/publisher/FluxWindowPredicate.java

```diff
@@ -824,54 +804,22 @@ public int requestFusion(int requestedMode) {
 		}
 
 		@Override
-		public boolean isCancelled() {
-			return cancelled;
-		}
-
-		@Override
-		public boolean isStarted() {
-			return once == 1 && !done && !cancelled;
-		}
-
-		@Override
-		public boolean isTerminated() {
-			return done;
-		}
-
-		@Override
-		public Throwable getError() {
-			return error;
-		}
-
-		@Override
-		public Object downstream() {
-			return actual;
-		}
-
-		@Override
-		public Object upstream() {
-			return parent;
-		}
-
-		@Override
-		public long getCapacity() {
-			WindowPredicateMain<T> parent = this.parent;
-			return parent != null ? parent.prefetch : -1L;
-		}
-
-		@Override
-		public long getPending() {
-			return queue == null || done ? -1L : queue.size();
-		}
-
-		@Override
-		public long requestedFromDownstream() {
-			return requested;
-		}
-
-		@Override
-		public long expectedFromUpstream() {
-			return produced;
+		public Object scan(Attr key) {
+			switch (key) {
+				case PARENT:
+					return parent;
+				case CANCELLED:
+					return cancelled;
+				case TERMINATED:
+					return done;
+				case BUFFERED:
+					return queue.size();
+				case ERROR:
+					return error;
+				case REQUESTED_FROM_DOWNSTREAM:
+					return requested;
+			}
+			return InnerOperator.super.scan(key);
```

Comment:
missing `EXPECTED_FROM_UPSTREAM` (was `produced`)
missing equivalent of `getCapacity`